odl/contrib/fom/supervised.py

```diff
@@ -394,67 +394,54 @@ def blurring(data, ground_truth, mask=None, normalized=False,
+def false_structures_mask(foreground, smoothness_factor=None):
+    """Return mask empasizing areas outside ``foreground``.
 
     Parameters
     ----------
-    data : `FnBaseVector`
-        Input data to compare to the ground truth.
-    ground_truth : `FnBaseVector`
-        Reference to compare ``data`` to.
-    mask : `array-like`, optional
-        Binary mask to define ROI in which FOM evaluation is performed.
-    normalized  : bool, optional
-        Boolean flag to switch between unormalized and normalized FOM.
+    foreground : `FnBaseVector`
+        The region that should be de-emphasised
     smoothness_factor : float, optional
-        Positive real number. Higher value gives smoother weighting.
+        Positive real number. Higher value gives smoother transition
+        between foreground and its complement.
 
     Returns
     -------
-    fs : float
-        FOM value, where a lower value means a better match.
-
-    Notes
-    -----
-    The FOM evaluates
-
-    .. math::
-        \mathrm{FS} = \\big \| \\alpha^{-1} (f - g) \\big \|^2_2,
-
-    or, in normalized form
+    result : `FnBaseVector`
+         Euclidean distances of elements in ``foreground``.
 
-    .. math::
-        \mathrm{FS_N} = \\frac{\\big \| \\alpha^{-1}(f - g)\\big \|^2_2}
-                              {\\big \| \\alpha^{-1} f \\big \|^2_2 +
-                               \\big \| \\alpha^{-1} g \\big \|^2_2}.
+    Examples
+    --------
+    >>> space = odl.uniform_discr(0, 1, 5)
+    >>> foreground = space.element([0, 0, 1.0, 0, 0])
+    >>> mask = false_structures_mask(foreground)
+    >>> np.asarray(mask)
+    array([ 0.4,  0.2,  0. ,  0.2,  0.4])
 
-    The weighting function :math:`\\alpha` is given as
+    Raises
+    ------
+    ValueError
+        If foreground is all zero or contains values not in {0, 1}.
 
-    .. math::
-        \\alpha(x) = e^{-\\frac{1}{k} \\beta_m(x)},
+    Notes
+    -----
+    This helper function computes the euclidean distance transform from each
+    point in ``foreground.space`` to ``foreground``.
 
-    where :math:`\\beta_m(x)` is the Euclidian distance transform of a
-    given binary mask :math:`m`, and :math:`k` positive real number that
-    controls the smoothness of the weighting function :math:`\\alpha`.
-    The weighting gives higher values to structures outside the region
-    of interest defined by the mask.
+    The weighting gives higher values to structures outside the foreground
+    as defined by the mask.
     """
     from scipy.ndimage.morphology import distance_transform_edt
+    space = foreground.space
 
-    if smoothness_factor is None:
-        smoothness_factor = np.mean(data.space.shape) / 10
-
-    if mask is not None:
-        mask = distance_transform_edt(1 - mask)
-        mask = np.exp(mask / smoothness_factor)
+    unique = np.unique(foreground)
+    if not np.array_equiv(unique, [0., 1.]):
+        raise ValueError('`foreground` contains values not in [0, 1] or has '
```

Comment:
I read `[0, 1]` as an interval. Also this restriction is nowhere mentioned. I'd use the term "binary mask".